daemon/container.go

```diff
@@ -261,45 +268,46 @@ func validateContainerConfig(config *containertypes.Config) error {
 	return validateHealthCheck(config.Healthcheck)
 }
 
-func validateHostConfig(hostConfig *containertypes.HostConfig) error {
+func validateHostConfig(hostConfig *containertypes.HostConfig) ([]string, error) {
```

Comment:
I'm usually not a fan of named output-variables, but I think this is a case where it helps understand the returned variables. But _don't_ name the `error` return; we try to do so only if the error-return is "special" (e.g. handled in a defer);

```suggestion
func validateHostConfig(hostConfig *containertypes.HostConfig) (warnings []string, _ error) {
```